modules/javafx.graphics/src/main/java/com/sun/prism/j2d/PrismPrintPipeline.java

```diff
@@ -116,17 +118,93 @@ public synchronized ObservableSet<Printer> getAllPrinters() {
             for (int i=0; i<allServices.length;i++) {
                 if (defService != null && defService.equals(allServices[i])) {
                     printers.add(defPrinter);
+                    pMap.put(defService, defPrinter);
                 } else {
-                    PrinterImpl impl = new J2DPrinter(allServices[i]);
-                    Printer printer = PrintHelper.createPrinter(impl);
-                    impl.setPrinter(printer);
-                    printers.add(printer);
+                    addNew(allServices[i], printers);
+                }
+            }
+            printerSet = FXCollections.observableSet(printers);
+            returnedPrinterSet =
+                FXCollections.unmodifiableObservableSet(printerSet);
+        } else {
+            PrintService[] newServices =
+                    PrintServiceLookup.lookupPrintServices(null, null);
+            if ((newServices.length != printerSet.size()) ||
+                    (lastTime + 120000) > System.currentTimeMillis()) {
```

Comment:
I missed this earlier. The test is backwards. It should be `<` not `>` .

Reply:
fixed